server/pulp/server/managers/content/orphan.py

```diff
@@ -220,6 +221,7 @@ def delete_orphans_by_type(content_type_id, content_unit_ids=None):
             if content_unit_ids is not None and content_unit['_id'] not in content_unit_ids:
                 continue
 
+            OrphanManager.delete_catalog_entry(content_unit['_id'], content_type_id)
```

Comment:
same suggestion here as below, to just do the delete using mongoengine's API instead of using a custom but equivalent API.